src/main/java/com/microsoft/alm/authentication/KeychainSecurityCliStore.java

```diff
@@ -304,6 +307,22 @@ else if ("blob".equals(type.toString()))
         // TODO: else if ("sint32".equals(type))
     }
 
+    public boolean isKeychainAvailable()
+    {
+        final String targetName = "test:isKeychainAvailable";
+        final Token token = new Token("this is a test token", TokenType.Test);
+        try
+        {
+            writeToken(targetName, token);
+        }
+        catch (final SecurityException e)
+        {
+            return false;
+        }
+        delete(targetName);
+        return true;
+    }
```

Comment:
> In isKeychainAvailable(), is there a better way to check availability of the keychain than an attempted write and delete?

In one of the early releases, I had noticed the desktop terminal had the `DISPLAY` environment variable set while the remote shells did not, until one of the users said it wasn't working on their system because, it turns out, that environment variable only is there if you have the _X Window System_ emulator/adapter installed.

I briefly considered checking for the presence of a `SSH_TTY` environment variable, but that's an indirect measurement and there could be other situations where that test doesn't work, either because we're in a different kind of remote shell or the keychain has already been unlocked.

Looking at other `security` commands, it appears `show-keychain-info` has no side-effects, yet exits with code `36` when "User interaction is not allowed."; I'll try using that.  Another benefit of _that_ command is then the issue around `delete()` goes away, plus the tests would be simpler!